### ISSUE TITLE
Helicopter noise tweetbot

### ISSUE THREAD

**Issue by imjasonh** [imjasonh/ideas#94]
I apparently live under a very common flight path for helicopters flying from Manhattan to the Hamptons, a fact that's especially apparent on Fridays and Sundays in the summer when there seem to be a lot of them going out or back. It wouldn't be a real problem except they fly really low, and pretty fast, and they're really loud. I have no idea whether they're at the legal altitude or whatever, but I do know they're loud and annoying.

Whenever I hear a particularly low/loud one, I open https://globe.adsbexchange.com/ on my phone to find it, and they're sometimes as low as 700 ft, going 120 knots (138 mph for land folk). [Here](https://user-images.githubusercontent.com/210737/127159024-65ec3994-b8c8-492e-88b4-9933140b9802.PNG)'s one going 140 mph, 375 feet above my head. [This page](https://aerocorner.com/blog/why-are-helicopters-so-loud/) says that's 75-87 dB from the ground, [not harmful except for long periods](https://www.cdc.gov/nceh/hearing_loss/what_noises_cause_hearing_loss.html), but, quote, "you may feel annoyed by the noise." Can confirm.

There's a [page to report helicopter to the FAA](https://portal.311.nyc.gov/article/?kanumber=KA-02267), but honestly who expects that to lead to anything.

> ...surprisingly, there is no comprehensive data on traffic in city airspace, which is completely unregulated below 1,200 feet.

(from _[There’s No One in Charge of New York’s Noisy Helicopters](https://www.curbed.com/2021/05/nyc-helicopter-noise-complaints.html)_)

Let's see if we can collect some data then, even if it's just over our small high-traffic slice of Brooklyn. [ADSB has an API](https://rapidapi.com/adsbx/api/adsbexchange-com1/) that's available for $10/month, with an endpoint to list aircraft within 1 nautical mile (1.15 land miles, wtf just use land miles ya boatnerds). If I can set up an RPi on the roof with a microphone that sends some signal when it detects noise over some dB level and triggers a request to list nearby helicopters, I can see if I can find one that's flying low and fast and grab its tail number. It'd be a good opportunity to learn some stuff.

With all the hard part out of the way, it'd be relatively easy to have a Twitter bot tweet the tail number, altitude, speed, noise level from the ground, and a little picture showing it on a map, maybe a recording of the noise. If I can get it to tweet while the thing's still overhead, followers can have push notifications tell them what's overhead while it's still there.

One complication is that I also live near a not-quiet expressway, so I'd need to make sure I don't get false alarms from horns honking, sirens, etc. But as a guess if it's 70+ dB for more than 2 seconds and there's a low-flying helicopter nearby, it's the helicopter's noise.

**Comment by vijayp** [comment]
you can probably get a directional microphone so that you can ensure the sound only comes from above? and/or use doppler shift to ensure that it's a helicopter moving in the same direction (and speed/height) as the aircraft data.

**Comment by imjasonh** [comment]
Yeah I was thinking of also trying to limit to frequencies that correlate with helicopters based on experience. I don't think I need to train an ML model but who knows.

In the meantime I was thinking, since I want a preview-tweet option that gets pushed to me anyway to confirm it's a helicopter. At that point, as a super lazy (boring) option, I might as well just use my own expertly trained human ears to detect the helicopter anyway, and just open a webpage when I hear one, see the preview tweet, and hit send.
